samples/java/appengine-java11/helloworld-servlet/README.md

```diff
@@ -0,0 +1,142 @@
+# Using the Java Agent on a Hello World Servlet on GAE Standard with Java 11
+
+> **Note**
+> This example was copied from
+[appengine-java11/helloworld-servlet](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/appengine-java11/helloworld-servlet) and modified for Snapshot Debugger Java agent use.
+
+The Java 11 runtime requires that your application have a `Main` class that
+starts a web server. This sample is dependent on artifact
+[`appengine-simple-jetty-main`](../appengine-simple-jetty-main) to provide a
+`Main` class that starts an embedded Jetty server that loads a WAR file.
+
+This example will use the App Engine Plugin to first build and stage the
+application. It will make some custom changes to the staging directory to add
+files before deploying it.
+
+It will also demonstrate the use of the `cdbg_extra_class_path` option for the
+Snapshot Debugger Java Agent.
+
+## Setup
+
+See [Prerequisites](../README.md#Prerequisites).
+
+Ensure your current working directory is
+`samples/java/appengine-java11/helloworld-server`, as all following instructions
+assumes this.
+
+- Note the `appengine-simple-jetty-main` dependency:
+```
+<dependency>
+  <groupId>com.example.appengine.demo</groupId>
+  <artifactId>simple-jetty-main</artifactId>
+  <version>1</version>
+  <scope>provided</scope>
+</dependency>
+```
+> **Note**
+> This dependency needs to be installed locally.
+
+### Install dependency locally
+
+Move into the directory to install, and then move back.
+
+```
+pushd ../appengine-simple-jetty-main
+mvm install
+popd
+```
+
+## Package your app:
+
+First use the maven App Engine Plugin to build and stage the application.
+
+```
+mvn clean package appengine:stage
+```
+
+## Add the Snapshot Debugger Java Agent
+
+Per [App Engine Staging
+Directory](../README.md#app-engine-staging-directory-and-the-snapshot-debugger-java-agent)
+we add in the Snapshot Debugger Java Agent.
+
+```
+mkdir target/appengine-staging/cdbg
+wget -qO- https://github.com/GoogleCloudPlatform/cloud-debug-java/releases/latest/download/cdbg_java_agent_gce.tar.gz | tar xvz -C target/appengine-staging/cdbg
+```
+
+## Extract the WAR file
+
+In this sample app the WAR file gets extracted to a random temporary location by
+Jetty, the Snapshot Debugger Java Agent will not be able to locate the class
+file in order to set breakpoints. To enable it to do so, the WAR file needs to
+be extracted to a known location that the agent can be configured to search for.
```

Comment:
Does this mean that we're uploading a duplicate copy of the compiled code?

Reply:
It does. Based on how the sample is setup it was the only way to make things work. I'll come back to this example later on and see if I can change things to avoid this, but for now it still demonstrates how the debugger can be made to work.